File: app/src/main/java/br/com/pauloc/mercadin/common/ConfigSharedPreferences.java
package br.com.pauloc.mercadin.common;

import android.content.Context;
import android.content.SharedPreferences;

import br.com.pauloc.mercadin.R;

public class ConfigSharedPreferences {
    private SharedPreferences sharedPreferences;
    private Context context;

    public ConfigSharedPreferences(Context context){
        this.context=context;

        this.sharedPreferences=context.getSharedPreferences(context.getResources().getString(R.string.preferencia_login), Context.MODE_PRIVATE);
    }

    public void guardarPreferencia(boolean logado, String utilizador){
        SharedPreferences.Editor editor=sharedPreferences.edit();

        editor.putBoolean(context.getResources().getString(R.string.loginlogado), logado);
        editor.putString(context.getResources().getString(R.string.loginutilizador), utilizador);

        editor.commit();
    }

    public String obterPreferencia(){
        String utilizador;

        utilizador=sharedPreferences.getString(context.getResources().getString(R.string.loginutilizador), "");

        return utilizador;
    }
}

File: app/src/main/java/br/com/pauloc/mercadin/common/ConfigSharedPreferences.kt
package br.com.pauloc.mercadin.common

import android.content.Context
import android.content.SharedPreferences
import br.com.pauloc.mercadin.R

class ConfigSharedPreferences(private val context: Context) {
    private val sharedPreferences: SharedPreferences
    fun guardarPreferencia(logado: Boolean, utilizador: String?) {
        val editor = sharedPreferences.edit()
        editor.putBoolean(context.resources.getString(R.string.loginlogado), logado)
        editor.putString(context.resources.getString(R.string.loginutilizador), utilizador)
        editor.commit()
    }

    fun obterPreferencia(): String? {
        val utilizador: String?
        utilizador = sharedPreferences.getString(context.resources.getString(R.string.loginutilizador), "")
        return utilizador
    }

    init {
        sharedPreferences = context.getSharedPreferences(context.resources.getString(R.string.preferencia_login), Context.MODE_PRIVATE)
    }
}